backend/parser.py

```diff
@@ -163,23 +164,91 @@ def parse(self, file):
 
             line = f.readline()
             while line:
-                matched_name = re.search(self.regex['sensor_name'], line)
-                matched_id = re.search(self.regex['sensor_id'], line)
-                
-                if matched_name and matched_id:
-                    matched_name = matched_name.group(0)
-                    matched_id = matched_id.group(0)
-
-                    samples[matched_id] = Sample(matched_name, matched_id)
+                if self.read_header(line, samples):
                     line = f.readline()
                 else:
                     break
 
-        return samples
+            # Read data fields and add to the corresponding sample object.
+            while line:
+                self.read_body(line, samples)
+                line = f.readline()
+
+        # Return only the values since the keys are no longer relevant.
+        return list(samples.values())
+
+    def read_header(self, line, samples) -> bool:
+        """Parses a header line to determine sensor name and id.
+            Creates new Sample objects for sensors.
+
+        Args:
+            line: The file line to be read.
+            samples: The dict of samples to add to when a new Sample is created.
+
+        Returns:
+            True if name and id data was parsed from the line.
+            False if the line did not contain data.
+        """
+
+        matched_name = re.search(self.regex['sensor_name'], line)
+        matched_id = re.search(self.regex['sensor_id'], line)
+
+        if matched_name and matched_id:
+            matched_name = matched_name.group(0)
+            matched_id = matched_id.group(0)
+
+            samples[matched_id] = Sample(matched_name, matched_id)
+            return True
+        else:
+            return False
+
+    def read_body(self, line, samples: dict):
+        """Reads a line from a file and parses relevant data fields from it.
+
+        Args:
+            line: The line from a file to be read.
+            samples: A dict with keys: sensor_id and values: Sample objects.
+        """
+        
+        # Determine this sensors ID.
+        if 'inline_id' in self.regex:
+            search = re.search(self.regex['inline_id'], line)
+            if search:
+                this_id = search.group(0)
+        else:
+            this_id = samples.keys()[0]
```

Comment:
I think you might hit an error here if the file does not contain a header, because no entry will be added, thus no keys.